tests/unit/amo/components/TestRatingsByStar.js

```diff
@@ -213,34 +227,65 @@ describe(__filename, () => {
       1: 0,
     };
     const addon = addonForGrouping(grouping);
+    const clientApp = CLIENT_APP_FIREFOX;
+    const lang = 'en-US';
+
+    dispatchClientMetadata({ clientApp, lang, store });
     store.dispatch(setGroupedRatings({ addonId: addon.id, grouping }));
+
     const root = render({
       addon,
+      history: fakeHistory,
       location: createFakeLocation({ query: { src } }),
+      store,
     });
-    const counts = root.find('.RatingsByStar-count').find(Link);
-
-    function validateLink(link, score) {
-      expect(link).toHaveProp(
-        'to',
-        reviewListURL({ addonSlug: addon.slug, score, src }),
+    const tableRow = root.find('.RatingByStar-table-row');
+
+    function validateLink(row, score) {
+      row.simulate('click', createFakeEvent());
+
+      sinon.assert.calledWith(
+        fakeHistory.push,
+        `/${lang}/${clientApp}${reviewListURL({
+          addonSlug: addon.slug,
+          score,
+          src,
+        })}`,
       );
     }
 
-    validateLink(counts.at(0), '5');
-    validateLink(counts.at(1), '4');
-    validateLink(counts.at(2), '3');
-    validateLink(counts.at(3), '2');
-    validateLink(counts.at(4), '1');
+    validateLink(tableRow.at(0), '5');
+    validateLink(tableRow.at(1), '4');
+    validateLink(tableRow.at(2), '3');
+    validateLink(tableRow.at(3), '2');
+    validateLink(tableRow.at(4), '1');
   });
 
   it('renders IconStar', () => {
     const root = render();
     const star = root.find(IconStar).at(0);
+
     expect(star).toHaveLength(1);
     expect(star).toHaveProp('selected');
   });
 
+  it('renders when star count is undefined', () => {
+    let fiveStarCount;
+    const grouping = {
+      5: fiveStarCount,
+      4: 1,
+      3: 0,
+      2: 0,
+      1: 0,
+    };
+    const addon = addonForGrouping(grouping);
+    store.dispatch(setGroupedRatings({ addonId: addon.id, grouping }));
+    const root = render({ addon });
+    const bar = root.find('.RatingsByStar-bar');
+
+    expect(bar.at(0).children()).toHaveLength(0);
```

Comment:
It would be nice to also have an assert for one of the bars that is not `undefined` which shows that `children()` has a length of `1`, so we can see that this test is in fact testing what we mean to test.

Reply:
Ohk! Nice, I will add this too!